### PR TITLE
Resolves FoamVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the FoamVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        SubscribeLocalEvent<FoamVisualizerComponent, ComponentInit>(OnInit);
+    }
+
+    private void OnInit(EntityUid uid, FoamVisualizerComponent comp, ComponentInit args)
+    {
+        var flick = new AnimationTrackSpriteFlick();
+        flick.LayerKey = FoamVisualLayers.Base;
+        flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(comp.State, 0f));
+        comp.Animation = new Animation {Length = TimeSpan.FromSeconds(comp.Delay)};
+        comp.Animation.AnimationTracks.Add(flick);
```
Comment (AJCM-git) [inline Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs:21]: make this a single object initializer

File: Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+[Access(typeof(FoamVisualizerSystem))]
+public sealed class FoamVisualizerComponent : Component
+{
+    public const string AnimationKey = "foamdissolve_animation";
+
+    [DataField("animationTime")]
+    public float Delay = 0.6f;
+
+    [DataField("animationState")]
+    public string State = "foam-dissolve";
```
Comment (AJCM-git) [inline Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs:14]: make this a mapped layer or a SpriteSpecifier

File: Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+namespace Content.Client.Chemistry.Visualizers;
+
+[RegisterComponent]
+[Access(typeof(FoamVisualizerSystem))]
+public sealed class FoamVisualizerComponent : Component
+{
+    public const string AnimationKey = "foamdissolve_animation";
+
+    [DataField("animationTime")]
+    public float Delay = 0.6f;
```
Comment (AJCM-git) [inline Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs:11]: rename this to AnimationTime

File: Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Chemistry.Visualizers;
+
+public sealed class FoamVisualizerSystem : VisualizerSystem<FoamVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FoamVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs:12]: rename the handler to OnComponentInit

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs (hunk 1 above)
Comment (TemporalOroboros) [reply Content.Client/Chemistry/Visualizers/FoamVisualizerSystem.cs:21]: I'm not entirely certain what you mean by this.

File: Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs:14]: Animations do not function with sprite specifiers as far as I can tell.

Comment (TemporalOroboros) [reply Content.Client/Chemistry/Visualizers/FoamVisualizerComponent.cs:14]: Wait, NVM the mapping is just for the animation.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**